[BUG] genRandomIP: 生成v6地址/掩码位64位时rand.Int入参为负

### Kube-OVN Version

v1.13.0

### Kubernetes Version

v1.25.3

### Operation-system/Kernel Version

"Ubuntu 22.04.4 LTS" 6.5.0-35-generic

### Description

unit test 入参：2001:db8::/64
![image](https://github.com/kubeovn/kube-ovn/assets/52371592/f2bb08d9-8383-4f82-b393-832a008981fd)



### Steps To Reproduce

获取随机 v6 ip 的 cidr掩码位为64（65位就没问题）

### Current Behavior

panic: crypto/rand: argument to Int is <= 0 [recovered]
        panic: crypto/rand: argument to Int is <= 0

### Expected Behavior

cidr掩码位为64的v6地址能生成随机v6地址

--- a/pkg/util/net.go
+++ b/pkg/util/net.go
@@ -200,7 +200,7 @@ func genRandomIP(cidr string, isIPv6 bool) string {
 	if isIPv6 {
 		hostBits = 128 - netMask
 	}
-	add, err := rand.Int(rand.Reader, big.NewInt(1<<(uint(hostBits)-1)))
+	add, err := rand.Int(rand.Reader, new(big.Int).Lsh(big.NewInt(1), uint(hostBits)-1))
 	if err != nil {
 		LogFatalAndExit(err, "failed to generate random ip")
 	}

--- a/pkg/util/net_test.go
+++ b/pkg/util/net_test.go
@@ -446,6 +446,51 @@ func TestGenerateRandomV4IP(t *testing.T) {
 	}
 }
 
+func TestGenerateRandomV6IP(t *testing.T) {
+	tests := []struct {
+		name     string
+		cidr     string
+		wantErr  bool
+		wantIPv6 bool
+	}{
+		{
+			name:     "valid IPv6 CIDR",
+			cidr:     "2001:db8::/64",
+			wantErr:  false,
+			wantIPv6: true,
+		},
+		{
+			name:     "invalid CIDR format",
+			cidr:     "2001:db8::1",
+			wantErr:  true,
+			wantIPv6: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ip := GenerateRandomV6IP(tt.cidr)
+			if tt.wantErr {
+				if ip != "" {
+					t.Errorf("GenerateRandomV6IP(%s) = %s; want empty string", tt.cidr, ip)
+				}
+				return
+			}
+
+			parsedIP, _, err := net.ParseCIDR(ip)
+			if err != nil {
+				t.Errorf("GenerateRandomV6IP(%s) returned invalid IP: %v", tt.cidr, err)
+				return
+			}
+
+			isIPv6 := parsedIP.To4() == nil
+			if isIPv6 != tt.wantIPv6 {
+				t.Errorf("GenerateRandomV6IP(%s) returned %v; want IPv6: %v", tt.cidr, parsedIP, tt.wantIPv6)
+			}
+		})
+	}
+}
+
 func TestIPToString(t *testing.T) {
 	tests := []struct {
 		name string

